components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/AbstractComplexController.java

```diff
@@ -114,14 +114,17 @@
     public void writeJSON(Patient patient, JSONObject json, Collection<String> selectedFieldNames)
     {
         PatientData<T> data = patient.getData(getName());
+        String jsonPropertyName = getJsonPropertyName();
+
         if (data == null) {
```

Comment:
it seems we still wont add anything to generated JSON if `data == null` but `selectedFieldNames` has a field this controller controls (as opposed to adding empty values that we should in this case)